lib/spack/spack/bootstrap/core.py

```diff
@@ -559,6 +559,21 @@ def ensure_patchelf_in_path_or_raise() -> spack.util.executable.Executable:
         )
 
 
+def ensure_winsdk_or_raise() -> None:
+    """Ensure the Windows SDK + WGL are available on system"""
+    externals = spack.detection.by_path(["win-sdk", "wgl"])
+    if not externals:
+        raise RuntimeError(
+            "Unable to find the Windows SDK, please install via the Visual Studio installer\
+before proceeding with Spack or provide the path to a non standard install via\
+'spack external find --path'"
+        )
```

Comment:
WGL is installed as part of the SDK (I don't think you can get the SDK without WGL), finding them both is more because we model both as separate packages, it should not be possible to detect only one.
Maybe I should fail with a different error if both are not found?

Reply:
Update: it is not possible to get the SDK without WGL unless you remove WGL manually, which is possible, I'll update the check to handle this.

Reply:
This is more about whether our *detection* fails to return both: even if they both have to be installed together on Windows, the function looking for them both might only return one (because of issues in our implementation)